### ISSUE TITLE
NameError: global name '__jid_event__' is not defined when running a runner in the scheduler

### ISSUE THREAD

**Issue by arthurlogilab** [saltstack/salt#44565]
### Description of Issue/Question

I am developing a custom runner which works fine when run using salt-run custom_runner.function

In it I have a progress message fired like in the documentation. 

```
        if display_progress:
            __jid_event__.fire_event({'message': 'A progress message'}, 'progress')
```

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

```
$ cat /etc/salt/master:
[snip]
schedule:
  custom_runner:
    function: custom_runner.func
    minutes: 1
$ salt-master -l debug 
[snip]
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/utils/schedule.py", line 847, in handle_func
    ret['return'] = self.functions[func](*args, **kwargs)
  File "/srv/salt/_runners/custom_runner.py", line 53, in func
    __jid_event__.fire_event({'message': 'A progress message'}, 'progress')
NameError: global name '__jid_event__' is not defined
```

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
# salt --versions-report
Salt Version:
           Salt: 2017.7.2
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 2.0.0
      gitpython: 2.1.1
          ioflo: Not Installed
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Jan 19 2017, 14:48:08)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 2.0.1
        timelib: Not Installed
        Tornado: 4.4.3
            ZMQ: 4.2.1
 
System Versions:
           dist: debian 9.2 
         locale: UTF-8
        machine: x86_64
        release: 4.9.0-4-amd64
         system: Linux
        version: debian 9.2 

```

**Comment by gtmanfred** [comment]
Gross,  I have fixed this here #44570

There may be more discussion on if there is a better way to fix this... Thanks! Daniel.

**Comment by gtmanfred** [comment]
This should be fixed, @Ch3LL can you close it?

Thanks,
Daniel
